### PR TITLE
bpo-39969: Remove ast.Param node class

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39969

### CODE REVIEW

File: Doc/whatsnew/3.9.rst [hunk 1]
```diff
@@ -667,6 +667,9 @@ Removed
 * The ``ast.Suite`` node class has been removed due to no longer being needed.
   (Contributed by Batuhan Taskaya in :issue:`39639`.)
 
+* The ``ast.Param`` node class has been removed due to no longer being needed.
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.9.rst]: It may be worth to merge this entry with the previous one.

Comment (isidentical) [reply Doc/whatsnew/3.9.rst]: Agreed

**Review by serhiy-storchaka — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]
:+1:

File: Python/compile.c [hunk 2]
```diff
@@ -5359,9 +5359,10 @@ compiler_subscript(struct compiler *c, expr_ty e)
         case AugStore:/* fall through to Store */
         case Store:   op = STORE_SUBSCR; break;
         case Del:     op = DELETE_SUBSCR; break;
-        case Param:
-            PyErr_SetString(PyExc_SystemError,
-                "param invalid in subscript expression");
+        default:
+            PyErr_Format(PyExc_SystemError,
+                         "expr_context kind %d should not be possible",
```
Comment (pablogsal) [inline Python/compile.c:5364]: I find the error message a bit odd. I would suggest changing it to something like:

"Received unexpected expr_context kind: %d"

Comment (isidentical) [reply Python/compile.c:5364]: This message is for consistency with other usages in the compiler. An example is mod_ty switch in compiler_mod

**Review by pablogsal — APPROVED** [review]

**Comment by asottile** [comment]
This is breaking pyflakes for what it's worth: https://travis-ci.org/github/PyCQA/pyflakes/jobs/663687707?utm_medium=notification&utm_source=email

**Pull request merged**